platform/platform-migratable/src/main/java/org/codice/ddf/platform/migratable/impl/PlatformMigratable.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.platform.migratable.impl;
+
+import static org.apache.commons.lang.Validate.notNull;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import javax.validation.constraints.NotNull;
+
+import org.codice.ddf.migration.AbstractMigratable;
+import org.codice.ddf.migration.Migratable;
+import org.codice.ddf.migration.MigrationException;
+import org.codice.ddf.migration.MigrationMetadata;
+import org.codice.ddf.migration.MigrationWarning;
+import org.codice.ddf.migration.util.MigratableUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class handles the export process for all Platform system files.
+ *
+ */
+public class PlatformMigratable extends AbstractMigratable implements Migratable {
+
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(PlatformMigratable.class);
+
+    private static final String KEYSTORE_SYSTEM_PROP = "javax.net.ssl.keyStore";
+
+    private static final String TRUSTSTORE_SYSTEM_PROP = "javax.net.ssl.trustStore";
+
+    private static final Path WS_SECURITY_DIR = Paths.get("etc", "ws-security");
+
+    private static final Path SYSTEM_PROPERTIES = Paths.get("etc", "system.properties");
+
+    private static final Path USERS_PROPERTIES = Paths.get("etc", "users.properties");
+    
+    private final MigratableUtil migratableUtil;
+    
+    public PlatformMigratable(@NotNull String description, boolean isOptional, @NotNull MigratableUtil migratableUtil) {
+        super(description, isOptional);
+        notNull(description, "description cannot be null.");
+        notNull(migratableUtil, "migrationUtil cannot be null.");
+        this.migratableUtil = migratableUtil;
+    }
+    
+    @Override
+    public MigrationMetadata export(Path exportPath) throws MigrationException {
+        notNull(exportPath, "exportPath cannot be null");
```

Comment:
We should change the `Migratable.export()` JavaDoc comment to say that `exportPath` will never be `null`. That way we don't have to check for this in every `Migratable` implementation.